tests/test_api.py

```diff
@@ -70,22 +70,23 @@ def test_send_new_token_response(client: Client):
     set_token_mock.assert_called_once_with("new-token", "local")
 
 
-# If ``sudo=True``, the X-Auth-Sudo header should be set with any value. This
-# test checks that the header is set.
-def test_send_sudo_header_set(client: Client):
-    client = Client(env="dev", region="bregenz.a1", sudo=True, _verify_ssl=False)
-    resp_data, errors = client.get("/test-x-sudo")
-    assert resp_data == {}
-    assert errors is None
+def test_send_sudo_header(client: Client):
+    for sudo in [True, False]:
+        client = Client(env="dev", region="bregenz.a1", sudo=sudo, _verify_ssl=False)
+        resp_data, errors = client.get("/client-headers")
+        assert (
+            "X-Auth-Sudo" in [key for key, value in resp_data]  # type: ignore
+        ) == sudo
 
 
-# If ``sudo=False``, no X-Auth-Sudo header should be set. This test checks that
-# the header is not set.
-def test_send_sudo_header_not_set(client: Client):
-    client = Client(env="dev", region="bregenz.a1", sudo=False, _verify_ssl=False)
-    resp_data, errors = client.get("/test-x-sudo")
-    assert resp_data == {"message": "Header not set, as expected."}
-    assert errors is None
+@pytest.mark.parametrize(
+    "argument,header", [(None, "bregenz.a1"), ("westeurope.azure", "westeurope.azure")]
+)
+def test_send_region_header(argument, header, client: Client):
+    client = Client(env="dev", region=argument, _verify_ssl=False)
+    resp_data, errors = client.get("/client-headers")
+    assert "X-Region" in [key for key, value in resp_data]  # type: ignore
+    assert dict(resp_data)["X-Region"] == header  # type: ignore
```

Comment:
When the `client_headers()` function below returns a jsonified dict instead of a list of 2-tuples, the assertion here and in `test_send_sudo_header` become simpler:
```python
return Response(json_data=dict(self.headers.items()))
```
```suggestion
    assert resp_data["X-Region"] == header
```
Using key access on the response data guarantees the item is in the response and that it has the given value.

Similarly, above, it'd be
```python
assert resp_data["X-Auth-Sudo"] == sudo
```